ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/block/BlockProcessorElectra.java

```diff
@@ -335,6 +344,105 @@ public void processDepositReceipts(
     }
   }
 
+  /*
+   Implements process_consolidation from consensus-specs (EIP-7251)
+  */
+  @Override
+  public void processConsolidations(
+      final MutableBeaconState state, final SszList<SignedConsolidation> consolidations)
+      throws BlockProcessingException {
+    final MutableBeaconStateElectra electraState = MutableBeaconStateElectra.required(state);
+
+    for (final SignedConsolidation signedConsolidation : consolidations) {
+      // If the pending consolidations queue is full, no consolidations are allowed in the block
+      assertCondition(
+          electraState.getPendingConsolidations().size()
+              < specConfigElectra.getPendingConsolidationsLimit(),
+          "Consolidation queue is full");
+
+      // If there is too little available consolidation churn limit, no consolidations are allowed
+      // in the block
+      assertCondition(
+          BeaconStateAccessorsElectra.required(beaconStateAccessors)
+              .getConsolidationChurnLimit(electraState)
+              .isGreaterThan(specConfigElectra.getMinActivationBalance()),
+          "Not enough consolidation churn limit available");
+
+      final Consolidation consolidation = signedConsolidation.getMessage();
+
+      // Verify that source != target, so a consolidation cannot be used as an exit.
+      assertCondition(
+          consolidation.getSourceIndex() != consolidation.getTargetIndex(),
+          "Consolidation source_index and target_index must be different");
+
+      final Validator sourceValidator = state.getValidators().get(consolidation.getSourceIndex());
+      final Validator targetValidator = state.getValidators().get(consolidation.getTargetIndex());
+      final UInt64 currentEpoch = beaconStateAccessors.getCurrentEpoch(electraState);
+
+      assertCondition(
+          predicatesElectra.isActiveValidator(sourceValidator, currentEpoch),
+          "Source validator in consolidation is inactive");
+      assertCondition(
+          predicatesElectra.isActiveValidator(targetValidator, currentEpoch),
+          "Target validator in consolidation is inactive");
+
+      // Verify exits for source and target have not been initiated
+      assertCondition(
+          sourceValidator.getExitEpoch().equals(FAR_FUTURE_EPOCH),
+          "Source validator in consolidation is exiting");
+      assertCondition(
+          targetValidator.getExitEpoch().equals(FAR_FUTURE_EPOCH),
+          "Target validator in consolidation is exiting");
+
+      // Consolidations must specify an epoch when they become valid; they are not valid before then
+      assertCondition(
+          currentEpoch.isGreaterThanOrEqualTo(consolidation.getEpoch()),
+          "Consolidation must have activation epoch in the future");
```

Comment:
This is more like
"Consolidation must have an activation epoch no greater than current epoch"